packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -564,7 +564,14 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
     protected async selectTerminalCwd(): Promise<string | undefined> {
         const roots = this.workspaceService.tryGetRoots();
         return this.quickPick.show(roots.map(
-            ({ uri }) => ({ label: this.labelProvider.getName(new URI(uri)), description: this.labelProvider.getLongName(new URI(uri)), value: uri })
+            ({ uri }) => {
+                const normalizedUri = new URI(uri).normalizePath();
```

Comment:
Maybe we should move into `DefaultLabelProvider` that it is applied for other clients as well?